packages/application-to-register-webapp/src/routes/developer/check-metric-file.js

```diff
@@ -0,0 +1,60 @@
+import constants from '../../utils/constants.js'
+import path from 'path'
+import { blobStorageConnector } from '@defra/bng-connectors-lib'
+
+const href = '#check-upload-correct-yes'
+const handlers = {
+  get: async (request, h) => {
+    const context = await getContext(request)
+    return h.view(constants.views.DEVELOPER_CHECK_UPLOAD_METRIC, context)
+  },
+  post: async (request, h) => {
+    const checkUploadMetric = request.payload.checkUploadMetric
+    const metricUploadLocation = request.yar.get(constants.redisKeys.METRIC_LOCATION)
+    request.yar.set(constants.redisKeys.METRIC_FILE_CHECKED, checkUploadMetric)
+    if (checkUploadMetric === 'no') {
+      // delete the file from blob storage
+      const config = {
+        containerName: 'trusted',
+        blobName: metricUploadLocation
+      }
+      await blobStorageConnector.deleteBlobIfExists(config)
+      request.yar.clear(constants.redisKeys.METRIC_LOCATION)
+      return h.redirect(constants.routes.DEVELOPER_UPLOAD_METRIC)
+    } else if (checkUploadMetric === 'yes') {
+      return h.redirect('/' + constants.views.DEVELOPER_CHECK_UPLOAD_METRIC, {
+        ...await getContext(request),
+        err: { text: '!TODO: Journey continuation not implemented' }
+      })
+    } else {
+      return h.view(constants.views.DEVELOPER_CHECK_UPLOAD_METRIC, {
+        filename: path.basename(metricUploadLocation),
+        ...await getContext(request),
+        err: [
+          {
+            text: 'Select yes if this is the correct file',
+            href
+          }
+        ]
+      })
+    }
+  }
+}
+
+const getContext = async request => {
```

Comment:
I know it's copy and paste from other check routes, but this is not an async function.